### PR TITLE
Add meson build scripts

### PR DESCRIPTION
**Pull request by clitic**
This PR adds [meson](https://github.com/mesonbuild/meson) build scripts.

Why to use meson in bento4?

- Makes it easier to use ap4 library through other meson projects
- Better install structure with customizable options
- Add support for compiling ap4 shared library
- Adds pkgconfig support
- Easier cross compilations

```powershell
meson setup builddir --buildtype=release --prefix=D:/bento4
meson install -C builddir
```

This is the default install structure.

```
+---bin
|       aac2mp4.exe
|       ...
|       mp4dash.bat
|       ...
|
+---docs
|       Bento4-HTML.zip
|       Bento4.chm
|       Bento4_SDK_documentation.doc
|       Bento4_SDK_documentation.pdf
|       LICENSE.txt
|
+---include
|   |
|   \---bento4
|          Ap4.h
|          Ap48bdlAtom.h
|           ...
|          Bento4C.h
|          ...
|
+---lib
|   |   libap4.a
|   |
|   \---pkgconfig
|           bento4.pc
|
\---utils
        aes.py
        check-indexes.py
        ...
```

Cross compiling through meson is also very easy. This example shows to compile using android ndk on windows machine.

**aarch64-linux-android31.ini**

```ini
[binaries]
c = 'aarch64-linux-android31-clang.cmd'
cpp = 'aarch64-linux-android31-clang++.cmd'
as = 'llvm-as'
ar = 'llvm-ar'
ld = 'ld'
ranlib = 'llvm-ranlib'
strip = 'llvm-strip'

[host_machine]
system = 'linux'
cpu_family = 'aarch64'
cpu = 'aarch64'
endian = 'little'

[built-in options] ; https://developer.android.com/ndk/guides/cpp-support
cpp_link_args = '-static-libstdc++'
```

```powershell
$env:PATH="D:/Android/android-ndk-r25b/toolchains/llvm/prebuilt/windows-x86_64/bin;$env:PATH"
meson setup builddir --buildtype=release --prefix=D:/bento4-android --cross-file=aarch64-linux-android31.ini
meson compile -C builddir
```

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+ap4_include = [
+  include_directories('Source/C++/Codecs'),
+  include_directories('Source/C++/Core'),
+  include_directories('Source/C++/Crypto'),
+  include_directories('Source/C++/MetaData')
+]
+
+if get_option('default_library') == 'both'
+  error('default library cannot be set to both, it can be either static or shared')
+endif
```
Comment (eli-schwartz) [inline meson.build:311]: This should only be true on Windows / with MSVC, probably?

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+)
+set_variable(meson.project_name() + '_dep', project_dep)
+
+install_headers(
+  ap4_headers + [
+    'Source/C++/Adapters/Ap4AtomixAdapters.h',
+    'Source/C++/Adapters/Ap4NeptuneAdapters.h',
+    'Source/C++/CApi/Bento4C.h'
+  ]
+)
```
Comment (eli-schwartz) [inline meson.build:336]: The cmake files install headers into the equivalent of adding `subdir: 'bento4'` here.

File: meson.build (hunk 1 above)
Comment (clitic) [reply meson.build]: I have tested to build with default library set to both and it works. I wasn't sure meson will link executables to ap4 shared library by default. Now I have fixed it

File: meson.build (hunk 2 above)
Comment (clitic) [reply meson.build]: Headers file should be kept under a subdirectory  bento4. I was trying to follow dir structure used [here](https://www.bento4.com/downloads). It is better to keep them in a separate directory. FIXED

**Review by eli-schwartz — APPROVED** [review]
Changes make sense to me.

**Pull request closed without merging**